src/test/testing_assert.cpp

```diff
@@ -94,88 +103,106 @@ ::testing::AssertionResult check_table_equal(const Table& tleft, const Table& tr
+    std::sort(expected_matrix.begin(), expected_matrix.end());
   }
 
-  for (unsigned row = 0; row < left.size(); row++)
-    for (ColumnID col{0}; col < left[row].size(); col++) {
-      if (is_null(left[row][col]) || is_null(right[row][col])) {
-        EXPECT_TRUE(is_null(left[row][col]) && is_null(right[row][col]));
-      } else if (tleft.column_type(col) == "float") {
-        auto left_val = type_cast<float>(left[row][col]);
-        auto right_val = type_cast<float>(right[row][col]);
+  for (unsigned row = 0; row < opossum_matrix.size(); row++)
+    for (ColumnID col{0}; col < opossum_matrix[row].size(); col++) {
+      if (is_null(opossum_matrix[row][col]) || is_null(expected_matrix[row][col])) {
+        EXPECT_TRUE(is_null(opossum_matrix[row][col]) && is_null(expected_matrix[row][col]));
+      } else if (opossum_table->column_type(col) == "float") {
+        auto left_val = type_cast<float>(opossum_matrix[row][col]);
+        auto right_val = type_cast<float>(expected_matrix[row][col]);
 
-        if (strict_types) {
-          EXPECT_EQ(tright.column_type(col), "float");
+        if (type_cmp_mode == TypeCmpMode::Strict) {
+          EXPECT_EQ(expected_table->column_type(col), "float");
         } else {
-          EXPECT_TRUE(tright.column_type(col) == "float" || tright.column_type(col) == "double");
+          EXPECT_TRUE(expected_table->column_type(col) == "float" || expected_table->column_type(col) == "double");
         }
-        EXPECT_NEAR(left_val, right_val, 0.0001) << "Row/Col:" << row << "/" << col;
-      } else if (tleft.column_type(col) == "double") {
-        auto left_val = type_cast<double>(left[row][col]);
-        auto right_val = type_cast<double>(right[row][col]);
+        if (float_comparison_mode == FloatComparisonMode::AbsoluteDifference) {
+          EXPECT_NEAR(left_val, right_val, 0.0001) << "Row/Col:" << row << "/" << col;
+        } else {
+          EXPECT_REL_NEAR(left_val, right_val, 0.0001) << "Row/Col:" << row << "/" << col;
+        }
+      } else if (opossum_table->column_type(col) == "double") {
+        auto left_val = type_cast<double>(opossum_matrix[row][col]);
+        auto right_val = type_cast<double>(expected_matrix[row][col]);
 
-        if (strict_types) {
-          EXPECT_EQ(tright.column_type(col), "double");
+        if (type_cmp_mode == TypeCmpMode::Strict) {
+          EXPECT_EQ(expected_table->column_type(col), "double");
+        } else {
+          EXPECT_TRUE(expected_table->column_type(col) == "float" || expected_table->column_type(col) == "double");
+        }
+        if (float_comparison_mode == FloatComparisonMode::AbsoluteDifference) {
+          EXPECT_NEAR(left_val, right_val, 0.0001) << "Row/Col:" << row << "/" << col;
         } else {
-          EXPECT_TRUE(tright.column_type(col) == "float" || tright.column_type(col) == "double");
+          EXPECT_REL_NEAR(left_val, right_val, 0.0001) << "Row/Col:" << row << "/" << col;
```

Comment:
I hate floats. :(

Reply:
who doesn't? .....and?

Reply:
...nothing. Just had to get that off my chest :D